frontend/src/routes/Upload.tsx

```diff
@@ -176,6 +196,22 @@ const UploadMain: React.FC<UploadMainProps> = ({ knownRoles, preselectedSeries }
 
     const progressHistory = useRef<ProgressHistory>([]);
     const abortController = useRef(new AbortController());
+    const [commit] = useMutation<UploadCreatePlaceholderMutation>(createPlaceholderMutation);
+
+    const createPlaceholder = async (event: PlaceholderEvent) => commit({
+        variables: {
+            event: {
+                ...event,
+                acl: [...event.acl].map(
+                    ([role, { actions }]) => ({
+                        role,
+                        actions: [...actions],
+                    }),
+                ),
```

Comment:
I've seen this transformation code before, are we sure there isn't already a util function somewhere we can use?